dll/cpl/usrmgr/userprops.c

```diff
@@ -87,80 +87,38 @@ SetUserProfileData(HWND hwndDlg,
-        pszProfilePath = HeapAlloc(GetProcessHeap(), 0, (nLength + 1) * sizeof(TCHAR));
-        GetDlgItemText(hwndDlg, IDC_USER_PROFILE_PATH, pszProfilePath, nLength + 1);
-        pUserInfo->usri3_profile = pszProfilePath;
-    }
+    pUserInfo->usri3_profile = GetDlgItemTextAlloc(hwndDlg, IDC_USER_PROFILE_PATH);
 
     /* Get the script path */
-    nLength = GetWindowTextLength(GetDlgItem(hwndDlg, IDC_USER_PROFILE_SCRIPT));
-    if (nLength == 0)
-    {
-        pUserInfo->usri3_script_path = NULL;
-    }
-    else
-    {
-        pszScriptPath = HeapAlloc(GetProcessHeap(), 0, (nLength + 1) * sizeof(TCHAR));
-        GetDlgItemText(hwndDlg, IDC_USER_PROFILE_SCRIPT, pszScriptPath, nLength + 1);
-        pUserInfo->usri3_script_path = pszScriptPath;
-    }
+    pUserInfo->usri3_script_path = GetDlgItemTextAlloc(hwndDlg, IDC_USER_PROFILE_SCRIPT);
 
     if (IsDlgButtonChecked(hwndDlg, IDC_USER_PROFILE_LOCAL) == BST_CHECKED)
     {
         /* Local home directory */
-        nLength = GetWindowTextLength(GetDlgItem(hwndDlg, IDC_USER_PROFILE_LOCAL_PATH));
-        if (nLength == 0)
-        {
-            pUserInfo->usri3_home_dir = NULL;
-        }
-        else
-        {
-            pszHomeDir = HeapAlloc(GetProcessHeap(), 0, (nLength + 1) * sizeof(TCHAR));
-            GetDlgItemText(hwndDlg, IDC_USER_PROFILE_LOCAL_PATH, pszHomeDir, nLength + 1);
-            pUserInfo->usri3_home_dir = pszHomeDir;
-        }
+        pUserInfo->usri3_home_dir = GetDlgItemTextAlloc(hwndDlg, IDC_USER_PROFILE_LOCAL_PATH);
     }
     else
     {
         /* Remote home directory */
-        nLength = GetWindowTextLength(GetDlgItem(hwndDlg, IDC_USER_PROFILE_REMOTE_PATH));
-        if (nLength == 0)
-        {
-            pUserInfo->usri3_home_dir = NULL;
-        }
-        else
-        {
-            pszHomeDir = HeapAlloc(GetProcessHeap(), 0, (nLength + 1) * sizeof(TCHAR));
-            GetDlgItemText(hwndDlg, IDC_USER_PROFILE_REMOTE_PATH, pszHomeDir, nLength + 1);
-            pUserInfo->usri3_home_dir = pszHomeDir;
-        }
+        pUserInfo->usri3_home_dir = GetDlgItemTextAlloc(hwndDlg, IDC_USER_PROFILE_REMOTE_PATH);
 
         nIndex = SendMessage(GetDlgItem(hwndDlg, IDC_USER_PROFILE_DRIVE), CB_GETCURSEL, 0, 0);
         if (nIndex != CB_ERR)
         {
-            nLength = SendMessage(GetDlgItem(hwndDlg, IDC_USER_PROFILE_DRIVE), CB_GETLBTEXTLEN, nIndex, 0);
-            pszHomeDrive = HeapAlloc(GetProcessHeap(), 0, (nLength + 1) * sizeof(TCHAR));
-            SendMessage(GetDlgItem(hwndDlg, IDC_USER_PROFILE_DRIVE), CB_GETLBTEXT, nIndex, (LPARAM)pszHomeDrive);
-            pUserInfo->usri3_home_dir_drive = pszHomeDrive;
+            pUserInfo->usri3_home_dir_drive =
+                GetComboBoxLBTextAlloc(hwndDlg, IDC_USER_PROFILE_DRIVE, nIndex);
+        }
+        else
+        {
+            pUserInfo->usri3_home_dir_drive = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY,
+                                                        sizeof(TCHAR));
```

Comment:
just place it in the same heapalloc line (and if you consider it too long then maybe put the whole HeapAlloc(....) call on a separate new line as you did for GetComboBoxLBTextAlloc above.

Reply:
Fixed.